x-pack/plugins/infra/public/pages/metrics/metric_detail/hooks/use_metrics_time.ts

```diff
@@ -64,9 +64,11 @@ export const useMetricsTime = () => {
     parseRange(urlState.time || DEFAULT_TIMERANGE)
   );
 
-  const updateTimeRange = useCallback((range: MetricsTimeInput) => {
+  const updateTimeRange = useCallback((range: MetricsTimeInput, parseDate = true) => {
```

Comment:
Nit: maybe we can always pass a parsed date to this function so here we can simplify the logic, wdyt?

Reply:
I don't want to change this function so much. We won't use it after this https://github.com/elastic/kibana/issues/164147 is done